### ISSUE TITLE
speed up iter_evoked?

### ISSUE THREAD

**Issue by jona-sassenhagen** [mne-tools/mne-python#6250]
I (mis)use epochs.iter_evokeds for confidence intervals from plot_compare_evokeds a lot. That is really slow. Is there some trivial way for speeding up iter_evokeds? (Instead of doing the sensible thing and speeding up plot_compare_evokeds itself ...)

I don't see one, it seems the problem is it's calling Epochs.__next__ and creating 100s of Evokeds via EvokedArray, which is simply inherently slow.

**Comment by mmagnuski** [comment]
I don't know how `iter_evokeds` works exactly but I assume it could create the Evoked object only once and only change the `.data` then (if it doesn't do this already).

**Comment by agramfort** [comment]
it's very possible it does a copy of the info which can take time. one
option would be to add a parameter to iter_evokeds method to avoid as many
copies as possible

**Comment by jona-sassenhagen** [comment]
Thanks, might be worth doing. I will self assign, although this is low priority I think because I'm abusing things here.

**Comment by agramfort** [comment]
see https://github.com/mne-tools/mne-python/pull/7164
